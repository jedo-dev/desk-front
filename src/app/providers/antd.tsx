import { ConfigProvider } from 'antd';
import React from 'react';

export const withAntdProvider = (component: () => React.ReactNode) => () =>
  (
    <ConfigProvider
      theme={{
        components: {
          Button: {
            colorPrimary: 'linear-gradient(90deg, rgba(116,207,159,1) 0%, rgba(49,189,185,1) 100%)',
            borderRadius: 50,
          },
        },
      }}>
      {component()}
    </ConfigProvider>
  );
