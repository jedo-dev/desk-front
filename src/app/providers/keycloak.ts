import Keycloak from 'keycloak-js';
const keycloak = new Keycloak({
  url: 'http://keycloak.udptest.cloudns.nz',
  realm: 'PABP',
  clientId: 'pabp-frontend-client',
});

export default keycloak;
