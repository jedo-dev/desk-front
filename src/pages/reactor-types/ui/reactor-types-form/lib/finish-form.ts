export const finishForm: (values: any) => void = (values) => {
  console.log('Введённые значения: ', values);
  return values;
};
