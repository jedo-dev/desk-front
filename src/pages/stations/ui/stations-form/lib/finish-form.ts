export const finishStationForm: (values: any) => void = (values) => {
  console.log('Введённые значения: ', values);
  return values;
};
