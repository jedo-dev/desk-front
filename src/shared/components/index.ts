export { AppLayout } from './app-layout';
export { BusinessProcessCardHeader } from './business-process-card-header/BusinessProcessCardHeader';
export { ReusedCard } from './reused-card/ReusedCard';
export { ReusedForm } from './reused-form/ReusedForm';
export { ReusedModalWithForm } from './reused-modal-with-form/ReusedModalWithForm';
export { ReusedTable } from './reused-table/ReusedTable';

//slice
export {
  resetFormItems,
  reusedFormReducer,
  setFormItems,
} from './reused-form/slice/reused-form-slice';
export {
  reusedModalWithFormReducer,
  setOpen,
} from './reused-modal-with-form/slice/reused-modal-with-form-slice';
export {
  reusedTableReducer,
  setActualRecord,
  setLoadingTable,
  setPaginationConfig,
  setTableFiltersConfig,
  setTableSortsConfig,
} from './reused-table/slice/reused-table';
