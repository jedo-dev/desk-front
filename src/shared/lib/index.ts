export { IconsMatcher } from './icons-matcher/icons-matcher';
export { MenuBuilder } from './menu-builder/menu-builder';
