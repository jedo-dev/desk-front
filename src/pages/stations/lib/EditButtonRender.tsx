import { useAppDispatch } from '@app/store/hooks/redux';
import { EditButton } from '@features';
import { setActualRecord, setLoadingTable, setOpen } from '@shared/components';
import { Flex } from 'antd';
import { AnyObject } from 'antd/es/_util/type';

export type editButtonRenderProps = {
  record: AnyObject;
};

export const EditButtonRender = ({ record }: editButtonRenderProps) => {
  const dispatch = useAppDispatch();

  return (
    <Flex gap='small' justify='center'>
      <EditButton
        dataTestId='clickEditTest'
        onClick={() => {
          dispatch(setActualRecord(record));
          dispatch(setOpen(true));

          // dispatch(setLoadingTable(false));
        }}
        showButton={true}
      />
    </Flex>
  );
};
