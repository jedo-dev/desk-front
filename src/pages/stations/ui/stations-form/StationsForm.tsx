import { ReusedModalWithForm, setFormItems } from '@shared/components';
import { useEffect } from 'react';
import { finishStationForm } from './lib/finish-form';
import { useAppDispatch, useAppSelector } from '@app/store/hooks/redux';

export const StationsForm = () => {
  const dispatch = useAppDispatch();
  const { openModal } = useAppSelector((state) => state.reusedModalWithFormReducer);
  const {
    actualRecord = {
      code_name: undefined,
      full_name: undefined,
      description: undefined,
      is_active: false,
    },
  } = useAppSelector((state) => state.reusedTableReducer);

  useEffect(() => {
    if (openModal === true) {
      dispatch(
        setFormItems([
          {
            type: 'inputTextArea',
            formItemProps: {
              mainProps: {
                name: 'code_name',
                label: 'Код станции',
                value: actualRecord['code_name'],
                rules: [
                  {
                    required: true,
                    message: 'Обязательное поле',
                  },
                ],
              },
            },
          },
          {
            type: 'inputTextArea',
            formItemProps: {
              mainProps: {
                name: 'full_name',
                label: 'Название станции',
                value: actualRecord['full_name'],
                rules: [
                  {
                    required: true,
                    message: 'Обязательное поле',
                  },
                ],
              },
            },
          },
          {
            type: 'inputTextArea',
            formItemProps: {
              mainProps: {
                name: 'description',
                label: 'Описание',
                value: actualRecord['description'],
              },
            },
          },
          {
            type: 'checkBox',
            formItemProps: {
              mainProps: {
                name: 'is_active',
                label: '',
                value: actualRecord['is_active'],
                valuePropName: 'checked',
              },
            },
            checkBox: {
              defaultCheckBoxText: 'Активный',
            },
          },
        ]),
      );
    }
  }, [openModal]);

  return (
    <ReusedModalWithForm
      onFinishForm={finishStationForm}
      antFormFlexProps={{ vertical: true }}
      dataTestid='testForm'
    />
  );

  //пример контроля положения элементов в форме

  // useEffect(() => {
  //   dispatch(
  //     setFormItems([
  //       {
  //         type: 'inputTextArea',
  //         formItemProps: {
  //           mainProps: {
  //             name: 'code_name',
  //             label: 'Код станции',
  //             value: (actualRecord && actualRecord['code_name']) || undefined,
  //             rules: [
  //               {
  //                 required: true,
  //                 message: 'Обязательное поле',
  //               },
  //             ],
  //           },
  //           style: {
  //             flexGrow: 1,
  //           },
  //         },
  //       },
  //       {
  //         type: 'inputTextArea',
  //         formItemProps: {
  //           mainProps: {
  //             name: 'full_name',
  //             label: 'Название станции',
  //             value: (actualRecord && actualRecord['full_name']) || undefined,
  //             rules: [
  //               {
  //                 required: true,
  //                 message: 'Обязательное поле',
  //               },
  //             ],
  //           },
  //           style: {
  //             flexGrow: 1,
  //           },
  //         },
  //       },
  //       {
  //         type: 'inputTextArea',
  //         formItemProps: {
  //           mainProps: {
  //             name: 'description',
  //             label: 'Описание',
  //             value: (actualRecord && actualRecord['description']) || undefined,
  //           },
  //           style: {
  //             width: '100%',
  //           },
  //         },
  //       },
  //       {
  //         type: 'checkBox',
  //         formItemProps: {
  //           mainProps: {
  //             name: 'is_active',
  //             label: '',
  //             value: (actualRecord && actualRecord['is_active']) || false,
  //             valuePropName: 'checked',
  //           },
  //           style: {
  //             display: 'flex',
  //             alignSelf: 'flex-end',
  //             justifyContent: 'flex-end',
  //           },
  //         },
  //         checkBox: {
  //           defaultCheckBoxText: 'Активный',
  //         },
  //       },
  //     ]),
  //   );
  // }, [open === true]);

  // return <ReusedModalWithForm open={open} onFinishForm={onFinishForm} />;
};
