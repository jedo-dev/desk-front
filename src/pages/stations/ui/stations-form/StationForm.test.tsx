import '@testing-library/jest-dom';
import { matchMediaConfig } from '@shared/test-configs/window-match-media';
import { describe, expect, it } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import { StationsForm } from './StationsForm';
import { withReduxProvider } from '@app/providers/redux';

import { setOpen } from '@shared/components';
import { finishStationForm } from './lib/finish-form';
import { useAppDispatch } from '@app/store/hooks/redux';

matchMediaConfig;

//компонент с хуком useAppDispatch()
const WrappedStationsForm = () => {
  const dispatch = useAppDispatch();
  dispatch(setOpen(true)); // Вызываем dispatch внутри компонента

  return <StationsForm />;
};

type stationFormItems = {
  code_name: string;
  full_name: string;
  description?: string;
  is_active?: boolean;
};

describe('StationsForm', () => {
  it('форма открыта и конфигурация элементов формы соответсвует требованиям макета', () => {
    render(withReduxProvider(() => <WrappedStationsForm />)());
    expect(screen.getByTestId('testForm')).toMatchSnapshot();
  });

  it('форма открыта и содержит поля на основе конфига', () => {
    render(withReduxProvider(() => <WrappedStationsForm />)());
    expect(screen.getByText('Код станции')).toBeInTheDocument();
    expect(screen.getByText('Название станции')).toBeInTheDocument();
    expect(screen.getByText('Описание')).toBeInTheDocument();
    expect(screen.getByText('Активный')).toBeInTheDocument();
  });

  it('при заполненных обязательных полях вызывается функция onFinish', async () => {
    render(withReduxProvider(() => <WrappedStationsForm />)());

    fireEvent.change(screen.getByLabelText('Код станции'), {
      target: { value: 'тест код станции' },
    });
    fireEvent.change(screen.getByLabelText('Название станции'), {
      target: { value: 'тест название станции' },
    });

    const mokInputObj: stationFormItems = {
      code_name: screen.getByLabelText('Код станции').innerHTML,
      full_name: screen.getByLabelText('Название станции').innerHTML,
    };

    const saveButton = screen.getByText('Сохранить');
    fireEvent.click(saveButton);

    expect(finishStationForm(mokInputObj)).toStrictEqual({
      code_name: 'тест код станции',
      full_name: 'тест название станции',
    });
  });
});
