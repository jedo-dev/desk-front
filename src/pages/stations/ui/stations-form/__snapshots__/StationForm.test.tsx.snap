// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StationsForm > форма открыта и конфигурация элементов формы соответсвует требованиям макета 1`] = `
<div
  class="ant-modal-root css-dev-only-do-not-override-1qhpsh8"
  data-testid="testForm"
>
  <div
    class="ant-modal-mask"
  />
  <div
    class="ant-modal-wrap"
    tabindex="-1"
  >
    <div
      aria-labelledby="test-id"
      aria-modal="true"
      class="ant-modal css-dev-only-do-not-override-1qhpsh8 ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
      role="dialog"
      style="width: 500px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; height: 0px; overflow: hidden; outline: none;"
        tabindex="0"
      />
      <div
        class="ant-modal-content"
      >
        <button
          aria-label="Close"
          class="ant-modal-close"
          type="button"
        >
          <span
            class="ant-modal-close-x"
          >
            <span
              aria-label="close"
              class="anticon anticon-close ant-modal-close-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="close"
                fill="currentColor"
                fill-rule="evenodd"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
                />
              </svg>
            </span>
          </span>
        </button>
        <div
          class="ant-modal-header"
        >
          <div
            class="ant-modal-title"
            id="test-id"
          >
            Редактирование элемента справочника
          </div>
        </div>
        <div
          class="ant-modal-body"
        >
          <form
            autocomplete="off"
            class="ant-form ant-form-vertical ant-form-middle css-dev-only-do-not-override-1qhpsh8"
          >
            <div
              class="ant-flex css-dev-only-do-not-override-1qhpsh8 ant-flex-wrap-wrap ant-flex-align-stretch ant-flex-gap-middle ant-flex-vertical"
            >
              <div
                class="ant-form-item css-dev-only-do-not-override-1qhpsh8"
                style="margin: 0px;"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1qhpsh8"
                >
                  <div
                    class="ant-col ant-form-item-label ant-form-item-label-left ant-form-item-label-wrap css-dev-only-do-not-override-1qhpsh8"
                  >
                    <label
                      class="ant-form-item-required ant-form-item-no-colon"
                      for="code_name"
                      title="Код станции"
                    >
                      Код станции
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1qhpsh8"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <textarea
                          aria-required="true"
                          class="ant-input css-dev-only-do-not-override-1qhpsh8 ant-input-outlined"
                          id="code_name"
                          placeholder="Введите значение"
                          style="resize: none; height: 2px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-form-item css-dev-only-do-not-override-1qhpsh8"
                style="margin: 0px;"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1qhpsh8"
                >
                  <div
                    class="ant-col ant-form-item-label ant-form-item-label-left ant-form-item-label-wrap css-dev-only-do-not-override-1qhpsh8"
                  >
                    <label
                      class="ant-form-item-required ant-form-item-no-colon"
                      for="full_name"
                      title="Название станции"
                    >
                      Название станции
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1qhpsh8"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <textarea
                          aria-required="true"
                          class="ant-input css-dev-only-do-not-override-1qhpsh8 ant-input-outlined"
                          id="full_name"
                          placeholder="Введите значение"
                          style="resize: none; height: 2px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-form-item css-dev-only-do-not-override-1qhpsh8"
                style="margin: 0px;"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1qhpsh8"
                >
                  <div
                    class="ant-col ant-form-item-label ant-form-item-label-left ant-form-item-label-wrap css-dev-only-do-not-override-1qhpsh8"
                  >
                    <label
                      class="ant-form-item-no-colon"
                      for="description"
                      title="Описание"
                    >
                      Описание
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1qhpsh8"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <textarea
                          class="ant-input css-dev-only-do-not-override-1qhpsh8 ant-input-outlined"
                          id="description"
                          placeholder="Введите значение"
                          style="resize: none; height: 2px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-form-item css-dev-only-do-not-override-1qhpsh8 ant-form-item-has-success"
                style="margin: 0px;"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-1qhpsh8"
                  value="false"
                >
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-1qhpsh8"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <label
                          class="ant-checkbox-wrapper ant-checkbox-wrapper-in-form-item css-dev-only-do-not-override-1qhpsh8"
                        >
                          <span
                            class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1qhpsh8"
                          >
                            <input
                              class="ant-checkbox-input"
                              data-testid="CreateNodeElementsCheckBox"
                              defaultcheckboxtext="Активный"
                              id="is_active"
                              type="checkbox"
                            />
                            <span
                              class="ant-checkbox-inner"
                            />
                          </span>
                          <span>
                            <span>
                              Активный
                            </span>
                          </span>
                        </label>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </form>
        </div>
        <div
          class="ant-modal-footer"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-1qhpsh8 ant-btn-default"
            type="button"
          >
            <span>
              Отмена
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1qhpsh8 ant-btn-primary"
            type="button"
          >
            <span>
              Сохранить
            </span>
          </button>
        </div>
      </div>
      <div
        aria-hidden="true"
        style="width: 0px; height: 0px; overflow: hidden; outline: none;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;
